### PR TITLE
Todo list

### PR DESCRIPTION
**Pull request by albert-sun**
Added task list functionality including creating new task, displaying current tasks, and sorting. Will crerate new branch for implementing material design after this.

### CODE REVIEW

File: react-groupup/src/UserDash.js [hunk 1]
```diff
@@ -1,47 +1,218 @@
...
+                alreadyContains = true;
+            }
+        });
+
+        if (!alreadyContains && description.value !== "") {
+            todoReference.add(newTask);
+        }
+
+        // reset form on completion
+        document.getElementById("create-task-form").reset();
```
Comment (upadhyan) [inline react-groupup/src/UserDash.js:75]: why is there a getElementByID?  should be using state

File: react-groupup/src/UserDash.js [hunk 2]
```diff
@@ -1,47 +1,218 @@
...
+            {/* Creating task from parameters */}
+            <div className="create-task" style={{margin: 10, textAlign: "center"}}>
+                <form onSubmit={createTask} id="create-task-form">
+                    <label style={{fontWeight: "bold"}}>New task</label> <br/>
+                    <label>Description: <input type="text" name="description"/> </label> <br/>
+                        <label style={{paddingRight: 10}}>Priority: <select name="priority" defaultValue={priorities[0]}>
+                            {/* Map priorities for easy modification */}
+                            {
+                                priorities.map((priority) =>
+                                    <option value={priority}>{priority}</option>
```
Comment (upadhyan) [inline react-groupup/src/UserDash.js:165]: This line is throwing an error

**Comment by albert-sun** [comment]
Massive implementation of task creation, listing, sorting, and deletion. TO-DO migrate to Material UI.
Fixed keys error and made form reset react-friendly.

**Comment by albert-sun** [comment]
Massive implementation of task creation, listing, sorting, and deletion. TO-DO migrate to Material UI.
Fixed keys error and made form reset react-friendly.

**Pull request merged**